pkg/cidata/schema.go

```diff
@@ -0,0 +1,40 @@
+package cidata
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+	"gopkg.in/yaml.v3"
+)
+
+const schemaURL = "https://raw.githubusercontent.com/canonical/cloud-init/main/cloudinit/config/schemas/schema-cloud-config-v1.json"
+
+//go:embed schema-cloud-config-v1.json
+var schemaText string
```

Comment:
I don't have time for a full review right now, but it bugs me that the URL points to `HEAD` of `main` in a repo, but we vendor a specific version. Shouldn't the URL match the version we embed?

I'm also not a big fan of manually vendoring files from other repos; I would rather create a git submodule for the other repo to create a fixed dependency to a specific commit.

I haven't tested it, but I assume we could make `schema-cloud-config-v1.json` a symlink into the submodule, a Go would still embed it correctly.

I know some people hate using submodules, so not sure if we want to go that route. Any opinions @AkihiroSuda ?

Reply:
I don't feel that using submodules is the best approach just for a single JSON file.

Probably we can just copy the JSON file into the repo?

Reply:
> Probably we can just copy the JSON file into the repo?

How do we even know which version of `cloud-init` this file comes from? It is just `HEAD` of `main` at some random point in time.

The name `schema-cloud-config-v1.json` is misleading, as it seems to imply it is a fixed version, but it it being updated all the time.

Eventually we will want to update it. What is the procedure to do it?

Reply:
`go generate` should wget it and validate the sha256sum?

Reply:
I can add a specific version of this file, using "main" was just what https://www.schemastore.org/json/ said

Reply:
https://raw.githubusercontent.com/canonical/cloud-init/24.1/cloudinit/config/schemas/schema-cloud-config-v1.json